goldenmaster/modules/tb_same1_module/tb_same1/generated/mqtt/sameenum1interfaceclient.cpp

```diff
@@ -0,0 +1,147 @@
+        m_publisher->publishProp1Changed(prop1);
+    }
+}
+
+Enum1Enum SameEnum1InterfaceClient::getProp1() const
+{
+    return m_data.m_prop1;
+}
+
+Enum1Enum SameEnum1InterfaceClient::func1(Enum1Enum param1)
+{
+    if(m_client == nullptr) {
+        return Enum1Enum::value1;
+    }
+    Enum1Enum value(func1Async(param1).get());
+    return value;
+}
+
+std::future<Enum1Enum> SameEnum1InterfaceClient::func1Async(Enum1Enum param1)
+{
+    if(m_client == nullptr) {
+        throw std::runtime_error("Client is not initialized");
+    }
+    return std::async(std::launch::async, [this,
+                    param1]()
+        {
+            std::promise<Enum1Enum> resultPromise;
+            static const auto topic = ApiGear::MQTT::Topic("tb.same1","SameEnum1Interface",ApiGear::MQTT::Topic::TopicType::Operation,"func1");
+            static const auto responseTopic = ApiGear::MQTT::Topic(topic.getEncodedTopic() + "/" + m_client->getClientId() + "/result");
+            ApiGear::MQTT::InvokeReplyFunc responseHandler = [&resultPromise](ApiGear::MQTT::InvokeReplyArg arg) {
+                const Enum1Enum& value = arg.value.get<Enum1Enum>();
+                resultPromise.set_value(value);
+            };
+            auto responseId = registerResponseHandler(responseHandler);
+            m_client->invokeRemote(topic, responseTopic,
+                nlohmann::json::array({param1}).dump(), responseId);
+            return resultPromise.get_future().get();
+        }
+    );
+}
+
+void SameEnum1InterfaceClient::onSignal(const ApiGear::MQTT::Topic& topic, const std::string& args, const ApiGear::MQTT::Topic&, const std::string&)
+{
+    nlohmann::json json_args = nlohmann::json::parse(args);
+    if(topic.getEntityName() == "sig1") {
+        m_publisher->publishSig1(json_args[0].get<Enum1Enum>());
+        return;
+    }
+}
+
+void SameEnum1InterfaceClient::onPropertyChanged(const ApiGear::MQTT::Topic& topic, const std::string& args, const ApiGear::MQTT::Topic&, const std::string&)
+{
+    nlohmann::json json_args = nlohmann::json::parse(args);
+    const std::string& name = topic.getEntityName();
+    applyState({ {name, json_args} });
+    return;
+}
+
+int SameEnum1InterfaceClient::registerResponseHandler(ApiGear::MQTT::InvokeReplyFunc handler)
+{
+    auto responseId = 0;
+    std::uniform_int_distribution<> distribution (0, 100000);
+    m_responseHandlerMutex.lock();
+    do {
+        responseId = distribution(randomNumberGenerator);
+    } while (m_responseHandlerMap.find(responseId) != m_responseHandlerMap.end());
+    m_responseHandlerMap.insert(std::pair<int, ApiGear::MQTT::InvokeReplyFunc>(responseId, handler));
+    m_responseHandlerMutex.unlock();
+
+    return responseId;
+}
+
+void SameEnum1InterfaceClient::onInvokeReply(const ApiGear::MQTT::Topic& /*topic*/, const std::string& args, const ApiGear::MQTT::Topic& /*responseTopic*/, const std::string& correlationData)
+{
+    const int randomId = std::stoi(correlationData);
```

Comment:
it is not randomID it is callbackId, we'd like to get callback for that exactly Id not a random one